src/v/cluster/rm_stm.cc

```diff
@@ -1103,7 +1103,15 @@ ss::future<result<kafka_result>> rm_stm::do_replicate(
 
 ss::future<> rm_stm::stop() {
     auto_abort_timer.cancel();
-    return raft::state_machine::stop();
+    return raft::state_machine::stop().then([this] {
+        for (const auto& entry : _log_state.seq_table) {
+            _log_state.unlink_lru_pid(entry.second);
+        }
+        vassert(
```

Comment:
I am not sure about assert there
Maybe vlog error, because it is `stop` and in general we can finish stopping redpanda process

Reply:
Thats an invariant that should always hold true. If not it can potentially crash. vassert() OTOH gives us much more meaningful crash diagnostics.